### PR TITLE
feat: add feature flag for displaying release notes sections (WPB-11702)

### PR DESCRIPTION
**Pull request by ohassine**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-11702" title="WPB-11702" target="_blank">StoryWPB-11702</a>  [Android] - Repoint New features link to https://social.intranet.bund.de/portal/wirebund
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

For some custom builds, we don't need to display the release notes section

### Solutions

Add a feature flag, so it can changed based on the custom build requirements

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_ 

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by damian-kaczmarek — CHANGES_REQUESTED** [review]
Wrong issue linked?

File: app/src/main/kotlin/com/wire/android/ui/home/whatsnew/WhatsNewScreen.kt [hunk 1]
```diff
@@ -74,13 +75,15 @@ fun WhatsNewScreenContent(
...
         modifier = modifier.fillMaxSize()
     ) {
-        folderWithElements(
-            items = buildList {
-                add(WhatsNewItem.WelcomeToNewAndroidApp)
-            },
-            onItemClicked = onItemClicked,
-            isLoading = false,
-        )
+        if (BuildConfig.SHOULD_DISPLAY_RELEASE_NOTES) {
```
Comment (MohamadJaara) [inline app/src/main/kotlin/com/wire/android/ui/home/whatsnew/WhatsNewScreen.kt:78]: question: this is also the task to change the all notes link, will this be in a following PR?

**Review by MohamadJaara — APPROVED** [review]
LGTM, only one question about another part of the task

**Review by damian-kaczmarek — APPROVED** [review]

File: app/src/main/kotlin/com/wire/android/ui/home/whatsnew/WhatsNewScreen.kt (hunk 1 above)
Comment (ohassine) [reply app/src/main/kotlin/com/wire/android/ui/home/whatsnew/WhatsNewScreen.kt:78]: Yes did the change here https://github.com/wireapp/android-customization/pull/30

**Pull request merged**